--- evolution-data-server-0.0.90.orig/calendar/libical/src/libical/icaltime.c	2004-01-09 15:55:41.000000000 +0100
+++ evolution-data-server/calendar/libical/src/libical/icaltime.c	2004-03-29 14:34:55.553360132 +0200
@@ -354,7 +354,7 @@ void unset_tz(char *tzstr)
  *	If the input timezone is null, no conversion is done; that is, the
  *	time is simply returned as time_t in its native timezone.
  */
-time_t icaltime_as_timet_with_zone(const struct icaltimetype tt,
+time_t icaltime_as_timet_with_zone(struct icaltimetype tt,
 	const icaltimezone *zone)
 {
     icaltimezone *utc_zone;
--- evolution-data-server-0.0.90.orig/calendar/libical/src/libical/icaltime.h	2003-10-21 20:28:27.000000000 +0200
+++ evolution-data-server/calendar/libical/src/libical/icaltime.h	2004-03-29 14:32:02.201881738 +0200
@@ -177,7 +177,7 @@ struct icaltimetype icaltime_from_week_n
 time_t icaltime_as_timet(const struct icaltimetype);
 
 /** Return the time as seconds past the UNIX epoch, using timezones. */
-time_t icaltime_as_timet_with_zone(const struct icaltimetype tt,
+time_t icaltime_as_timet_with_zone(struct icaltimetype tt,
 	const icaltimezone *zone);
 
 /** Return a string represention of the time, in RFC2445 format. The
