diff --git a/configure.ac b/configure.ac
index d38cd85..8a89b73 100644
--- a/configure.ac
+++ b/configure.ac
@@ -164,11 +164,7 @@ dnl ******************************
 dnl Compiler Warning Flags
 dnl ******************************
 AS_COMPILER_FLAGS(WARNING_FLAGS,
-	"-DG_DISABLE_DEPRECATED
-	-DPANGO_DISABLE_DEPRECATED
-	-DGDK_PIXBUF_DISABLE_DEPRECATED
-	-DGTK_DISABLE_DEPRECATED
-	-DG_DISABLE_SINGLE_INCLUDES
+	"-DG_DISABLE_SINGLE_INCLUDES
 	-DGTK_DISABLE_SINGLE_INCLUDES
 	-DGSEAL_ENABLE
 	-Wall -Wextra
