From f64da6b7b5f5f51d6c37a2c3dafc93c1f2bdd944 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Wed, 27 Aug 2014 15:12:58 +0200
Subject: Bug 734853 - CamelNetworkService fails to connect to 'localhost'

A regression after a change for bug 731585.

diff --git a/camel/camel-network-service.c b/camel/camel-network-service.c
index d1ad212..c54f284 100644
--- a/camel/camel-network-service.c
+++ b/camel/camel-network-service.c
@@ -665,7 +665,7 @@ network_service_new_connectable (CamelNetworkService *service)
 	host = camel_network_settings_dup_host_ensure_ascii (network_settings);
 	port = camel_network_settings_get_port (network_settings);
 
-	if (host && *host && g_ascii_strcasecmp (host, "localhost") != 0)
+	if (host && *host)
 		connectable = g_network_address_new (host, port);
 
 	g_free (host);
diff --git a/camel/camel-network-settings.c b/camel/camel-network-settings.c
index 9bd2c06..7322634 100644
--- a/camel/camel-network-settings.c
+++ b/camel/camel-network-settings.c
@@ -57,7 +57,7 @@ camel_network_settings_default_init (CamelNetworkSettingsInterface *iface)
 			"host",
 			"Host",
 			"Host name for the network service",
-			"localhost",
+			"",
 			G_PARAM_READWRITE |
 			G_PARAM_CONSTRUCT |
 			G_PARAM_STATIC_STRINGS));
diff --git a/camel/camel-sasl-digest-md5.c b/camel/camel-sasl-digest-md5.c
index 49846f9..53b835a 100644
--- a/camel/camel-sasl-digest-md5.c
+++ b/camel/camel-sasl-digest-md5.c
@@ -838,8 +838,10 @@ sasl_digest_md5_challenge_sync (CamelSasl *sasl,
 
 	g_return_val_if_fail (user != NULL, NULL);
 
-	if (host == NULL)
+	if (!host || !*host) {
+		g_free (host);
 		host = g_strdup ("localhost");
+	}
 
 	password = camel_service_get_password (service);
 	g_return_val_if_fail (password != NULL, NULL);
diff --git a/camel/camel-sasl-gssapi.c b/camel/camel-sasl-gssapi.c
index d8b53b4..5bf09a8 100644
--- a/camel/camel-sasl-gssapi.c
+++ b/camel/camel-sasl-gssapi.c
@@ -315,8 +315,10 @@ sasl_gssapi_challenge_sync (CamelSasl *sasl,
 
 	g_return_val_if_fail (user != NULL, NULL);
 
-	if (host == NULL)
+	if (!host || !*host) {
+		g_free (host);
 		host = g_strdup ("localhost");
+	}
 
 	switch (priv->state) {
 	case GSSAPI_STATE_INIT:
-- 
cgit v0.10.1

