From f64da6b7b5f5f51d6c37a2c3dafc93c1f2bdd944 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Wed, 27 Aug 2014 15:12:58 +0200
Subject: Bug 734853 - CamelNetworkService fails to connect to 'localhost'

A regression after a change for bug 731585.

--- evolution-data-server-3.12.5.orig/camel/camel-network-service.c
+++ evolution-data-server-3.12.5/camel/camel-network-service.c
@@ -665,7 +665,7 @@ network_service_new_connectable (CamelNe
 	host = camel_network_settings_dup_host (network_settings);
 	port = camel_network_settings_get_port (network_settings);
 
-	if (host && *host && g_ascii_strcasecmp (host, "localhost") != 0)
+	if (host && *host)
 		connectable = g_network_address_new (host, port);
 
 	g_free (host);
--- evolution-data-server-3.12.5.orig/camel/camel-network-settings.c
+++ evolution-data-server-3.12.5/camel/camel-network-settings.c
@@ -56,7 +56,7 @@ camel_network_settings_default_init (Cam
 			"host",
 			"Host",
 			"Host name for the network service",
-			"localhost",
+			"",
 			G_PARAM_READWRITE |
 			G_PARAM_CONSTRUCT |
 			G_PARAM_STATIC_STRINGS));
--- evolution-data-server-3.12.5.orig/camel/camel-sasl-digest-md5.c
+++ evolution-data-server-3.12.5/camel/camel-sasl-digest-md5.c
@@ -838,8 +838,10 @@ sasl_digest_md5_challenge_sync (CamelSas
 
 	g_return_val_if_fail (user != NULL, NULL);
 
-	if (host == NULL)
+	if (!host || !*host) {
+		g_free (host);
 		host = g_strdup ("localhost");
+	}
 
 	password = camel_service_get_password (service);
 	g_return_val_if_fail (password != NULL, NULL);
--- evolution-data-server-3.12.5.orig/camel/camel-sasl-gssapi.c
+++ evolution-data-server-3.12.5/camel/camel-sasl-gssapi.c
@@ -315,8 +315,10 @@ sasl_gssapi_challenge_sync (CamelSasl *s
 
 	g_return_val_if_fail (user != NULL, NULL);
 
-	if (host == NULL)
+	if (!host || !*host) {
+		g_free (host);
 		host = g_strdup ("localhost");
+	}
 
 	switch (priv->state) {
 	case GSSAPI_STATE_INIT:
