From d072dd312f50e3d3444bc14977f0cd2b022f9667 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Tue, 13 Dec 2011 12:09:35 +0000
Subject: Bug #652173 - EContact's file-as getter doesn't populate empty value

---
diff --git a/addressbook/libebook/e-contact.c b/addressbook/libebook/e-contact.c
index b9fdf41..f938dd8 100644
--- a/addressbook/libebook/e-contact.c
+++ b/addressbook/libebook/e-contact.c
@@ -566,7 +566,15 @@ static gpointer
 fileas_getter (EContact *contact,
                EVCardAttribute *attr)
 {
-	if (!attr) {
+	GList *p = NULL;
+
+	if (attr) {
+		p = e_vcard_attribute_get_values (attr);
+		if (!p || !p->data || !*((const gchar *) p->data))
+			p = NULL;
+	}
+
+	if (!p) {
 		/* Generate a FILE_AS field */
 		EContactName *name;
 		gchar *new_file_as = NULL;
@@ -609,7 +617,7 @@ fileas_getter (EContact *contact,
 	}
 
 	if (attr) {
-		GList *p = e_vcard_attribute_get_values (attr);
+		p = e_vcard_attribute_get_values (attr);
 
 		return p && p->data ? p->data : (gpointer) "";
 	} else {
--
cgit v0.9.0.2
