diff -ruN evolution-data-server-1.8.0.orig/camel/camel.pc.in evolution-data-server-1.8.0/camel/camel.pc.in
--- evolution-data-server-1.8.0.orig/camel/camel.pc.in	2004-12-02 09:03:30.000000000 +0100
+++ evolution-data-server-1.8.0/camel/camel.pc.in	2006-09-06 22:52:10.000000000 +0200
@@ -9,5 +9,6 @@
 Name: camel
 Description: the Evolution MIME message handling library
 Version: @VERSION@
-Libs: -L${libdir} -lcamel-1.2 -ledataserver-1.2
+Requires: glib-2.0, libedataserver-@API_VERSION@
+Libs: -L${libdir} -lcamel-@API_VERSION@
 Cflags: -I${privincludedir}
diff -ruN evolution-data-server-1.8.0.orig/camel/camel-provider.pc.in evolution-data-server-1.8.0/camel/camel-provider.pc.in
--- evolution-data-server-1.8.0.orig/camel/camel-provider.pc.in	2004-12-02 09:03:30.000000000 +0100
+++ evolution-data-server-1.8.0/camel/camel-provider.pc.in	2006-09-06 22:53:32.000000000 +0200
@@ -10,5 +10,6 @@
 Name: camel-provider
 Description: The Evolution mail access library
 Version: @VERSION@
-Libs: -L${libdir} -lcamel-1.2 -lcamel-provider-1.2 -ledataserver-1.2
+Requires: camel-@API_VERSION@
+Libs: -L${libdir} -lcamel-provider-@API_VERSION@
 Cflags: -I${privincludedir}
