diff -Nur evolution-data-server-1.10.1/addressbook/libebook/e-vcard.c evolution-data-server-1.10.1.new/addressbook/libebook/e-vcard.c
--- evolution-data-server-1.10.1/addressbook/libebook/e-vcard.c	2007-04-09 14:42:56.000000000 +0200
+++ evolution-data-server-1.10.1.new/addressbook/libebook/e-vcard.c	2007-05-10 08:19:55.000000000 +0200
@@ -807,7 +807,7 @@
 		GList *p;
 		EVCardAttribute *attr = l->data;
 		GString *attr_str;
-		int l;
+		glong len;
 
 		if (!g_ascii_strcasecmp (attr->name, "VERSION"))
 			continue;
@@ -883,21 +883,23 @@
 		 * When generating a content line, lines longer than 75
 		 * characters SHOULD be folded
 		 */
-		l = 0;
-		do {
-			
-			if ((g_utf8_strlen (attr_str->str, -1) -l) > 75) {
-				char *p;
-
-				l += 75;
-				p = g_utf8_offset_to_pointer (attr_str->str, l);
-			
-				g_string_insert_len (attr_str, (p - attr_str->str), CRLF " ", sizeof (CRLF " ") - 1);
-			}
-			else
-				break;
-		} while (l < g_utf8_strlen (attr_str->str, -1));
-
+		len = g_utf8_strlen (attr_str->str, -1);
+		if (len > 75) {
+			GString *fold_str = g_string_sized_new (attr_str->len + len/74*3);
+			gchar *pos1 = attr_str->str;
+			gchar *pos2 = pos1;
+			pos2 = g_utf8_offset_to_pointer (pos2, 75);
+
+			do {
+				g_string_append_len (fold_str, pos1, pos2 - pos1);
+				g_string_append (fold_str, CRLF " ");
+				pos1 = pos2;
+				pos2 = g_utf8_offset_to_pointer (pos2, 74);
+			} while (pos2 < attr_str->str + attr_str->len);
+			g_string_append (fold_str, pos1);
+			g_string_free (attr_str, TRUE);
+			attr_str = fold_str;
+		}
 		g_string_append (attr_str, CRLF);
 
 		g_string_append (str, attr_str->str);
