--- addressbook/libebook/e-book.c.orig	2007-04-22 22:31:25.263966048 +0200
+++ addressbook/libebook/e-book.c	2007-04-22 22:33:18.211024798 +0200
@@ -3639,7 +3639,7 @@
 	uid = gconf_client_get_string (gconf, SELF_UID_KEY, NULL);
 	g_object_unref (gconf);
 
-	rv = (uid && strcmp (uid, e_contact_get_const (contact, E_CONTACT_UID)));
+	rv = (uid && !strcmp (uid, e_contact_get_const (contact, E_CONTACT_UID)));
 
 	g_free (uid);
 
