diff -urN evolution-data-server-1.8.1.orig/servers/exchange/storage/exchange-account.c evolution-data-server-1.8.1/servers/exchange/storage/exchange-account.c
--- evolution-data-server-1.8.1.orig/servers/exchange/storage/exchange-account.c	2006-10-02 08:23:08.000000000 +0200
+++ evolution-data-server-1.8.1/servers/exchange/storage/exchange-account.c	2006-10-11 23:58:05.270413250 +0200
@@ -54,7 +54,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#define d(x)
+#define d(x) x
 #define ADS_UF_DONT_EXPIRE_PASSWORD 0x10000
 #define ONE_HUNDRED_NANOSECOND 0.000000100
 #define SECONDS_IN_DAY 86400
@@ -1244,9 +1244,7 @@
 					      account->priv->identity_name,
 					      account->priv->identity_email,
 					      account->priv->source_uri,
-					      account->priv->offline_sync);
-	d(g_print ("exchange-account.c:setup_account_hierarchies:offline_sync=%d\n", account->priv->offline_sync));
-
+					      TRUE);
 	setup_hierarchy (account, hier);
 	g_free (phys_uri_prefix);
 
@@ -1277,7 +1275,7 @@
 					      account->priv->identity_name,
 					      account->priv->identity_email,
 					      account->priv->source_uri,
-					      account->priv->offline_sync);
+					      TRUE);
 	setup_hierarchy (account, hier);
 	g_free (phys_uri_prefix);
 
