diff -ruN evolution-data-server-1.6.2.orig/configure.in evolution-data-server-1.6.2/configure.in
--- evolution-data-server-1.6.2.orig/configure.in	2006-05-29 15:16:16.000000000 +0200
+++ evolution-data-server-1.6.2/configure.in	2006-05-30 18:39:36.000000000 +0200
@@ -355,7 +355,7 @@
 	
 	if test "x${check_manually}" = "xno"; then
 		AC_MSG_CHECKING(Mozilla NSPR pkg-config module name)
-		mozilla_nspr_pcs="nspr mozilla-nspr firefox-nspr"
+		mozilla_nspr_pcs="nspr mozilla-nspr firefox-nspr xulrunner-nspr"
 		for pc in $mozilla_nspr_pcs; do
 			if $PKG_CONFIG --exists $pc; then
 				AC_MSG_RESULT($pc)
@@ -365,7 +365,7 @@
 		done
 		
 		AC_MSG_CHECKING(Mozilla NSS pkg-config module name)
-		mozilla_nss_pcs="nss mozilla-nss firefox-nss"
+		mozilla_nss_pcs="nss mozilla-nss firefox-nss xulrunner-nss"
 		for pc in $mozilla_nss_pcs; do
 			if $PKG_CONFIG --exists $pc; then
 				AC_MSG_RESULT($pc)
@@ -388,8 +388,8 @@
 			MANUAL_NSS_LIBS=""
 		else
 			check_manually="yes"
-			mozilla_nspr=""
-			mozilla_nss=""
+			mozilla_nspr="xulrunner-nspr"
+			mozilla_nss="xulrunner-nss"
 		fi
 	fi
 	
