diff -ruN evolution-data-server-2.22.0.orig/libedataserver/e-categories.c evolution-data-server-2.22.0/libedataserver/e-categories.c
--- evolution-data-server-2.22.0.orig/libedataserver/e-categories.c	2007-11-29 13:53:48.000000000 +0100
+++ evolution-data-server-2.22.0/libedataserver/e-categories.c	2008-03-11 21:01:54.000000000 +0100
@@ -30,6 +30,8 @@
 
 #include "libedataserver-private.h"
 
+#define d(x)
+
 typedef struct {
 	char *category;
 	char *icon_file;
@@ -162,7 +164,7 @@
 
 	filename = build_categories_filename ();
 
-	g_debug ("Saving categories to \"%s\"", filename);
+	d (g_debug ("Saving categories to \"%s\"", filename) );
 
 	/* build the file contents */
 	buffer = g_string_new ("<categories>\n");
@@ -268,7 +270,7 @@
 	if (!g_file_test (filename, G_FILE_TEST_EXISTS))
 		goto exit;
 
-	g_debug ("Loading categories from \"%s\"", filename);
+	d ( g_debug ("Loading categories from \"%s\"", filename) );
 
 	if (!g_file_get_contents (filename, &contents, &length, &error)) {
 		g_warning ("Unable to load categories: %s", error->message);
@@ -324,7 +326,7 @@
 	if (string == NULL || *string == '\0')
 		goto exit;
 
-	g_debug ("Loading categories from GConf key \"%s\"", key);
+	d (g_debug ("Loading categories from GConf key \"%s\"", key) );
 
 	n_added = parse_categories (string, strlen (string));
 	if (n_added == 0)
@@ -401,20 +403,20 @@
 
 	n_added = load_categories ();
 	if (n_added > 0) {
-		g_debug ("Loaded %d categories", n_added);
+		d ( g_debug ("Loaded %d categories", n_added) );
 		save_is_pending = FALSE;
 		return;
 	}
 
 	n_added = migrate_old_categories ();
 	if (n_added > 0) {
-		g_debug ("Loaded %d categories", n_added);
+		d ( g_debug ("Loaded %d categories", n_added) );
 		save_categories ();
 		return;
 	}
 
 	load_default_categories ();
-	g_debug ("Loaded default categories");
+	d ( g_debug ("Loaded default categories") );
 	save_categories ();
 }
 
