--- libedataserver/e-iconv.c.orig	2007-10-22 10:29:56.587916703 +0200
+++ libedataserver/e-iconv.c	2007-10-22 10:37:22.113305748 +0200
@@ -135,15 +135,18 @@
 	{ "euckr-0",        "EUC-KR"     },
 	{ "5601",           "EUC-KR"     },
 	{ "zh_TW-euc",      "EUC-TW"     },
-	{ "zh_CN.euc",      "gb2312"     },
+	{ "zh_CN.euc",      "gb18030"    },
 	{ "zh_TW-big5",     "BIG5"       },
-	{ "euc-cn",         "gb2312"     },
+	{ "euc-cn",         "gb18030"    },
 	{ "big5-0",         "BIG5"       },
 	{ "big5.eten-0",    "BIG5"       },
 	{ "big5hkscs-0",    "BIG5HKSCS"  },
-	{ "gb2312-0",       "gb2312"     },
-	{ "gb2312.1980-0",  "gb2312"     },
-	{ "gb-2312",        "gb2312"     },
+	/* Also treat gb2312 as gb18030 to cope with user agents that encode */
+	/* gb18030 characters as gb2312 (MS) */
+	{ "gb2312-0",       "gb18030"    },
+	{ "gb2312.1980-0",  "gb18030"    },
+	{ "gb-2312",        "gb18030"    },
+	{ "gb2312",         "gb18030"    },
 	{ "gb18030-0",      "gb18030"    },
 	{ "gbk-0",          "GBK"        },
 
