diff -ru evolution-data-server-1.6.1.orig/addressbook/backends/file/Makefile.am evolution-data-server-1.6.1/addressbook/backends/file/Makefile.am
--- evolution-data-server-1.6.1.orig/addressbook/backends/file/Makefile.am	2005-06-22 14:34:21.000000000 +0200
+++ evolution-data-server-1.6.1/addressbook/backends/file/Makefile.am	2006-05-02 12:30:27.000000000 +0200
@@ -4,7 +4,6 @@
 	-I$(top_builddir)				\
 	-I$(top_srcdir)/addressbook			\
 	-I$(top_builddir)/addressbook			\
-	-I$(top_srcdir)/libdb/dist			\
         $(EVOLUTION_ADDRESSBOOK_CFLAGS)
 
 extension_LTLIBRARIES = libebookbackendfile.la
@@ -18,7 +17,7 @@
 	$(top_builddir)/addressbook/libebook/libebook-1.2.la		\
 	$(top_builddir)/addressbook/libedata-book/libedata-book-1.2.la	\
 	$(top_builddir)/libedataserver/libedataserver-1.2.la		\
-	$(top_builddir)/libdb/dist/libdb-4.1.la				\
+	-ldb								\
 	$(EVOLUTION_ADDRESSBOOK_LIBS)
 
 libebookbackendfile_la_LDFLAGS =	\
diff -ru evolution-data-server-1.6.1.orig/configure.in evolution-data-server-1.6.1/configure.in
--- evolution-data-server-1.6.1.orig/configure.in	2006-04-10 18:58:13.000000000 +0200
+++ evolution-data-server-1.6.1/configure.in	2006-05-02 12:31:11.000000000 +0200
@@ -1276,7 +1276,6 @@
 export privincludedir
 export privdatadir
 AC_CONFIG_SUBDIRS(calendar/libical)
-AC_CONFIG_SUBDIRS(libdb/dist)
 
 AC_OUTPUT([
 Makefile
@@ -1334,7 +1333,6 @@
 camel/tests/stream/Makefile
 camel/camel.pc
 camel/camel-provider.pc
-libdb/Makefile
 libedataserver/Makefile
 libedataserver/libedataserver.pc
 libedataserverui/Makefile
@@ -1360,13 +1358,6 @@
 po/Makefile.in
 ])
 
-case $host in
-*-mingw*)
-	# Override the old libtool in libdb/dist with the one from here
-	cp libtool libdb/dist
-	;;
-esac
-
 echo "
 	evolution-data-server has been configured as follows:
 	LDAP support:     $msg_ldap"
diff -ru evolution-data-server-1.6.1.orig/libedataserver/Makefile.am evolution-data-server-1.6.1/libedataserver/Makefile.am
--- evolution-data-server-1.6.1.orig/libedataserver/Makefile.am	2006-01-10 08:59:04.000000000 +0100
+++ evolution-data-server-1.6.1/libedataserver/Makefile.am	2006-05-02 12:32:54.000000000 +0200
@@ -1,7 +1,6 @@
 INCLUDES =								\
 	-I$(top_srcdir)							\
 	-DG_LOG_DOMAIN=\"e-data-server\"				\
-	-I$(top_srcdir)/libdb/dist					\
 	-DE_DATA_SERVER_PREFIX=\"$(prefix)\"				\
 	-DEVOLUTION_LOCALEDIR=\""$(localedir)"\"			\
 	-DE_DATA_SERVER_EXTENSIONDIR=\"$(extensiondir)\"		\
@@ -49,7 +48,7 @@
 libedataserver_1_2_la_LIBADD = 				\
 	$(E_DATA_SERVER_LIBS)				\
 	$(ICONV_LIBS)					\
-	$(top_builddir)/libdb/dist/libdb-4.1.la		\
+	-ldb						\
 	$(SOCKET_LIBS)
 
 libedataserver_1_2_la_LDFLAGS = \
diff -ru evolution-data-server-1.6.1.orig/Makefile.am evolution-data-server-1.6.1/Makefile.am
--- evolution-data-server-1.6.1.orig/Makefile.am	2005-12-02 01:06:42.000000000 +0100
+++ evolution-data-server-1.6.1/Makefile.am	2006-05-02 12:31:58.000000000 +0200
@@ -1,7 +1,7 @@
 AUTOMAKE_OPTIONS = 1.6
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 
-SUBDIRS = libdb win32 libedataserver servers camel addressbook calendar libedataserverui src docs art po
+SUBDIRS = win32 libedataserver servers camel addressbook calendar libedataserverui src docs art po
 DIST_SUBDIRS= win32 libedataserver servers camel addressbook calendar libedataserverui src docs art po
 
 changelogs =			\
@@ -27,12 +27,9 @@
 	intltool-update.in				\
 	intltool-extract.in				\
 	evolution-data-server-zip.in			\
-	$(pkgconfig_DATA:-$(API_VERSION).pc=.pc.in) 	\
-	$(LIBDB_FILES)
+	$(pkgconfig_DATA:-$(API_VERSION).pc=.pc.in) 
 
 DISTCLEANFILES = intltool-extract intltool-merge intltool-update iconv-detect.h $(pkgconfig_DATA)
-distclean-local:
-	(cd libdb && $(MAKE) $(AM_MAKEFLAGS) distclean)
 
 LIBDB_FILES= \
 libdb/Makefile.am \
