diff -ruN evolution-data-server-1.8.0.orig/camel/Makefile.am evolution-data-server-1.8.0/camel/Makefile.am
--- evolution-data-server-1.8.0.orig/camel/Makefile.am	2006-05-26 13:15:07.000000000 +0200
+++ evolution-data-server-1.8.0/camel/Makefile.am	2006-09-06 23:03:19.000000000 +0200
@@ -281,7 +281,7 @@
 	camel-utf8.h				\
 	camel.h
 
-libcamel_1_2_la_LDFLAGS = -version-info 0:0:0 $(NO_UNDEFINED)
+libcamel_1_2_la_LDFLAGS = -version-info $(LIBCAMEL_CURRENT):$(LIBCAMEL_REVISION):$(LIBCAMEL_AGE) $(NO_UNDEFINED)
 
 libcamel_1_2_la_LIBADD =				\
 	$(top_builddir)/libedataserver/libedataserver-${API_VERSION}.la \
diff -ruN evolution-data-server-1.8.0.orig/camel/Makefile.in evolution-data-server-1.8.0/camel/Makefile.in
--- evolution-data-server-1.8.0.orig/camel/Makefile.in	2006-09-04 06:53:09.000000000 +0200
+++ evolution-data-server-1.8.0/camel/Makefile.in	2006-09-06 23:04:23.000000000 +0200
@@ -789,7 +789,7 @@
 	camel-utf8.h				\
 	camel.h
 
-libcamel_1_2_la_LDFLAGS = -version-info 0:0:0 $(NO_UNDEFINED)
+libcamel_1_2_la_LDFLAGS = -version-info $(LIBCAMEL_CURRENT):$(LIBCAMEL_REVISION):$(LIBCAMEL_AGE) $(NO_UNDEFINED)
 libcamel_1_2_la_LIBADD = \
 	$(top_builddir)/libedataserver/libedataserver-${API_VERSION}.la \
 	$(CAMEL_LIBS)							\
