diff --git a/evolution-data-server-2.27.91/configure b/evolution-data-server-2.27.91/configure
index c2668d6..299e2ff 100755
--- a/evolution-data-server-2.27.91/configure
+++ b/evolution-data-server-2.27.91/configure
@@ -24791,7 +24791,7 @@ if test $use_gweather = yes; then
 
 fi
 
-camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers
+camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers-`expr ${LIBCAMEL_CURRENT} - ${LIBCAMEL_AGE}`
 
 
 INTERFACE_VERSION="$BASE_VERSION"
diff --git a/evolution-data-server-2.27.91/configure.ac b/evolution-data-server-2.27.91/configure.ac
index c89caff..a9f324d 100644
--- a/evolution-data-server-2.27.91/configure.ac
+++ b/evolution-data-server-2.27.91/configure.ac
@@ -1586,7 +1586,7 @@ if test $use_gweather = yes; then
 fi
 
 dnl separate camel from e-d-s? or should it be under the same spot? same for now.
-camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers
+camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers-`expr ${LIBCAMEL_CURRENT} - ${LIBCAMEL_AGE}`
 AC_SUBST(camel_providerdir)
 
 dnl IDL/Component Versioning
