diff -ruN evolution-data-server-2.22.0.orig/configure evolution-data-server-2.22.0/configure
--- evolution-data-server-2.22.0.orig/configure	2008-03-09 18:55:24.000000000 +0100
+++ evolution-data-server-2.22.0/configure	2008-03-11 21:12:06.000000000 +0100
@@ -31861,7 +31861,7 @@
 weatherdatadir="$privdatadir/weather"
 
 
-camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers
+camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers-`expr ${LIBCAMEL_CURRENT} - ${LIBCAMEL_AGE}`
 
 
 
diff -ruN evolution-data-server-2.22.0.orig/configure.in evolution-data-server-2.22.0/configure.in
--- evolution-data-server-2.22.0.orig/configure.in	2008-03-09 18:18:32.000000000 +0100
+++ evolution-data-server-2.22.0/configure.in	2008-03-11 21:13:10.000000000 +0100
@@ -1519,7 +1519,8 @@
 AC_SUBST(weatherdatadir)
 
 dnl separate camel from e-d-s?  or should it be under the same spot?  same for now.
-camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers
+dnl add version to dir to avoid filename conflict between versions  
+camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers-`expr ${LIBCAMEL_CURRENT} - ${LIBCAMEL_AGE}`
 AC_SUBST(camel_providerdir)
 
 dnl ************************
