diff -ru evolution-data-server-1.6.2.orig/configure evolution-data-server-1.6.2/configure
--- evolution-data-server-1.6.2.orig/configure	2006-05-29 15:25:13.000000000 +0200
+++ evolution-data-server-1.6.2/configure	2006-05-30 19:06:13.000000000 +0200
@@ -30714,7 +30714,7 @@
 weatherdatadir="$privdatadir/weather"
 
 
-camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers
+camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers-$LIBCAMEL_CURRENT
 
 
 
diff -ru evolution-data-server-1.6.2.orig/configure.in evolution-data-server-1.6.2/configure.in
--- evolution-data-server-1.6.2.orig/configure.in	2006-05-30 18:59:48.000000000 +0200
+++ evolution-data-server-1.6.2/configure.in	2006-05-30 19:07:05.000000000 +0200
@@ -1281,7 +1281,8 @@
 AC_SUBST(weatherdatadir)
 
 dnl separate camel from e-d-s?  or should it be under the same spot?  same for now.
-camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers
+dnl add version to dir to avoid filename conflict between versions
+camel_providerdir='${libdir}'/evolution-data-server-$API_VERSION/camel-providers-$LIBCAMEL_CURRENT
 AC_SUBST(camel_providerdir)
 
 dnl ************************
