diff -u -r1.330 -r1.330.2.2
--- evolution-data-server-1.4.1.1.orig/camel/providers/imap/camel-imap-store.c	2005/08/31 04:26:02	1.330
+++ evolution-data-server.1.4.1.1/camel/providers/imap/camel-imap-store.c	2006/01/10 08:10:48	1.330.2.2
@@ -178,6 +178,7 @@
 camel_imap_store_finalize (CamelObject *object)
 {
 	CamelImapStore *imap_store = CAMEL_IMAP_STORE (object);
+	CamelDiscoStore *disco = CAMEL_DISCO_STORE (service);
 
 	/* This frees current_folder, folders, authtypes, streams, and namespace. */
 	camel_service_disconnect((CamelService *)imap_store, TRUE, NULL);
@@ -191,6 +192,12 @@
 		g_free (imap_store->base_url);
 	if (imap_store->storage_path)
 		g_free (imap_store->storage_path);
+
+	if (disco->diary) {
+		camel_object_unref (disco->diary);
+		disco->diary = NULL;
+	}
+
 }
 
 static void
@@ -1539,7 +1546,6 @@
 imap_disconnect_offline (CamelService *service, gboolean clean, CamelException *ex)
 {
 	CamelImapStore *store = CAMEL_IMAP_STORE (service);
-	CamelDiscoStore *disco = CAMEL_DISCO_STORE (service);
 
 	if (store->istream) {
 		camel_stream_close(store->istream);
@@ -1570,12 +1576,7 @@
 		g_free (store->namespace);
 		store->namespace = NULL;
 	}
-	
-	if (disco->diary) {
-		camel_object_unref (disco->diary);
-		disco->diary = NULL;
-	}
-	
+		
 	return TRUE;
 }
 
