diff -u -r1.71 -r1.71.2.1
--- evolution-data-server-1.4.1.1.orig/camel/providers/imap/camel-imap-command.c	2005/08/31 04:26:02	1.71
+++ evolution-data-server.1.4.1.1/camel/providers/imap/camel-imap-command.c	2005/12/01 07:33:52	1.71.2.1
@@ -824,8 +824,9 @@
 			if (imap_is_atom (string)) {
 				outptr += sprintf (outptr, "%s", string);
 			} else {
-				if (store->capabilities & IMAP_CAPABILITY_LITERALPLUS) {
-					outptr += sprintf (outptr, "{%d+}\r\n%s", (int)strlen(string), string);
+				len = strlen (string);
+				if (len && store->capabilities & IMAP_CAPABILITY_LITERALPLUS) {
+					outptr += sprintf (outptr, "{%d+}\r\n%s", len, string);
 				} else {
 					char *quoted = imap_quote_string (string);
 					
