diff -u -r1.71.2.1 -r1.71.2.2
--- evolution-data-server-1.4.1.1.orig/camel/providers/imap/camel-imap-command.c	2005/12/01 07:33:52	1.71.2.1
+++ evolution-data-server.1.4.1.1/camel/providers/imap/camel-imap-command.c	2005/12/15 05:16:23	1.71.2.2
@@ -766,11 +766,19 @@
 		case 'S':
 		case 'F':
 			string = va_arg (ap, char *);
+			/* NB: string is freed during output */
 			if (*p == 'F') {
-				/* NB: this is freed during output */
 				char *s = camel_imap_store_summary_full_from_path(store->summary, string);
-				string = s?s:camel_utf8_utf7(string);
+				if (s) {
+					string = camel_utf8_utf7(s);
+					g_free(s);
+				} else {
+					string = camel_utf8_utf7(string);
+				}
+			} else {
+				string = camel_utf8_utf7(string);
 			}
+				
 			arglen = strlen (string);
 			g_ptr_array_add (args, string);
 			if (imap_is_atom (string)) {
@@ -835,8 +843,7 @@
 				}
 			}
 			
-			if (*p == 'F')
-				g_free (string);
+			g_free (string);
 			break;
 		default:
 			*outptr++ = '%';
