diff -u -r1.16 -r1.16.2.1
--- evolution-data-server-1.4.1.1.orig/camel/providers/imap/camel-imap-store-summary.c	2005/08/31 04:26:02	1.16
+++ evolution-data-server.1.4.1.1/camel/providers/imap/camel-imap-store-summary.c	2005/12/15 05:16:23	1.16.2.1
@@ -195,7 +195,7 @@
 	} else
 		path = (char *)full_name;
 
-	return camel_utf7_utf8(path);
+	return g_strdup(path);
 }
 
 static guint32 hexnib(guint32 c)
@@ -273,7 +273,7 @@
 	camel_utf8_putc(&f, c);
 
 	/* merge old path part if required */
-	f = camel_utf8_utf7(full);
+	f = g_strdup(full);
 	if (si) {
 		full = g_strdup_printf("%s%s", camel_imap_store_info_full_name(s, si), f);
 		g_free(f);
