diff -u -r1.330.2.2 -r1.330.2.3
--- evolution-data-server-1.4.1.1.orig/camel/providers/imap/camel-imap-store.c	2006/01/10 08:10:48	1.330.2.2
+++ evolution-data-server.1.4.1.1/camel/providers/imap/camel-imap-store.c	2006/01/17 00:27:51	1.330.2.3
@@ -245,8 +245,9 @@
 	   CamelException *ex)
 {
 	CamelImapStore *imap_store = CAMEL_IMAP_STORE (service);
+	CamelDiscoStore *disco_store = CAMEL_DISCO_STORE (service);
 	CamelStore *store = CAMEL_STORE (service);
-	char *tmp;
+	char *tmp, *path;
 	CamelURL *summary_url;
 
 	CAMEL_SERVICE_CLASS (parent_class)->construct (service, session, provider, url, ex);
@@ -281,6 +282,11 @@
 	if (camel_url_get_param (url, "filter_junk_inbox"))
 		imap_store->parameters |= IMAP_PARAM_FILTER_JUNK_INBOX;
 
+	/* setup journal*/
+	path = g_strdup_printf ("%s/journal", imap_store->storage_path);
+	disco_store->diary = camel_disco_diary_new (disco_store, path, ex);
+	g_free (path);
+
 	/* setup/load the store summary */
 	tmp = alloca(strlen(imap_store->storage_path)+32);
 	sprintf(tmp, "%s/.ev-store-summary", imap_store->storage_path);
@@ -1442,7 +1448,7 @@
 			 * for the given path, even if that path doesn't exist.
 			 */
 			response = camel_imap_command (store, NULL, ex,
-						       "LIST %S \"\"",
+						       "LIST %G \"\"",
 						       store->namespace);
 		} else {
 			/* Plain IMAP4 doesn't have that idiom, so we fall back
@@ -1450,7 +1456,7 @@
 			 * the folder doesn't exist (eg, if namespace is "").
 			 */
 			response = camel_imap_command (store, NULL, ex,
-						       "LIST \"\" %S",
+						       "LIST \"\" %G",
 						       store->namespace);
 		}
 		if (!response)
@@ -1509,10 +1515,6 @@
 		store->refresh_stamp = time(0);
 	}
 	
-	path = g_strdup_printf ("%s/journal", store->storage_path);
-	disco_store->diary = camel_disco_diary_new (disco_store, path, ex);
-	g_free (path);
-	
  done:
 	/* save any changes we had */
 	camel_store_summary_save((CamelStoreSummary *)store->summary);
@@ -1530,11 +1532,7 @@
 {
 	CamelImapStore *store = CAMEL_IMAP_STORE (service);
 	CamelDiscoStore *disco_store = CAMEL_DISCO_STORE (service);
-	char *path;
 
-	path = g_strdup_printf ("%s/journal", store->storage_path);
-	disco_store->diary = camel_disco_diary_new (disco_store, path, ex);
-	g_free (path);
 	if (!disco_store->diary)
 		return FALSE;
 	
@@ -1624,15 +1622,10 @@
 	CamelImapResponse *response;
 	CamelFolder *current_folder;
 	
-	if (camel_disco_store_status (disco) != CAMEL_DISCO_STORE_ONLINE)
-		return;
-	
 	CAMEL_SERVICE_LOCK (imap_store, connect_lock);
 
-	if (!camel_imap_store_connected(imap_store, ex)) {
-		CAMEL_SERVICE_UNLOCK(imap_store, connect_lock);
-		return;
-	}
+	if (!camel_imap_store_connected(imap_store, ex)) 
+		goto done;
 
 	current_folder = imap_store->current_folder;
 	if (current_folder && imap_summary_is_dirty (current_folder->summary)) {
@@ -1644,6 +1637,7 @@
 			camel_imap_response_free (imap_store, response);
 	}
 	
+done:
 	CAMEL_SERVICE_UNLOCK (imap_store, connect_lock);
 }
 
@@ -1885,7 +1879,7 @@
 			guint32 flags;
 			int i;
 			
-			if (!(response = camel_imap_command (imap_store, NULL, ex, "LIST \"\" %S", parent_real))) {
+			if (!(response = camel_imap_command (imap_store, NULL, ex, "LIST \"\" %G", parent_real))) {
 				CAMEL_SERVICE_UNLOCK (imap_store, connect_lock);
 				g_free (parent_name);
 				g_free (parent_real);
@@ -1950,7 +1944,7 @@
 				
 				/* add the dirsep to the end of parent_name */
 				name = g_strdup_printf ("%s%c", parent_real, imap_store->dir_sep);
-				response = camel_imap_command (imap_store, NULL, ex, "CREATE %S",
+				response = camel_imap_command (imap_store, NULL, ex, "CREATE %G",
 							       name);
 				g_free (name);
 				
@@ -1969,7 +1963,7 @@
 		g_free (parent_name);
 
 		folder_real = camel_imap_store_summary_path_to_full(imap_store->summary, folder_name, imap_store->dir_sep);
-		response = camel_imap_command (imap_store, NULL, ex, "CREATE %S", folder_real);
+		response = camel_imap_command (imap_store, NULL, ex, "CREATE %G", folder_real);
 		if (response) {
 			camel_imap_store_summary_add_from_full(imap_store->summary, folder_real, imap_store->dir_sep);
 
@@ -2135,7 +2129,7 @@
 			if (imap_store->dir_sep == '.') {
 				CamelImapResponse *response;
 
-				response = camel_imap_command (imap_store, NULL, NULL, "RENAME %F %S", path, nfull);
+				response = camel_imap_command (imap_store, NULL, NULL, "RENAME %F %G", path, nfull);
 				if (response)
 					camel_imap_response_free (imap_store, response);
 			}
@@ -2253,7 +2247,7 @@
 	}
 
 	need_convert = FALSE;
-	response = camel_imap_command (imap_store, NULL, ex, "LIST \"\" %S",
+	response = camel_imap_command (imap_store, NULL, ex, "LIST \"\" %G",
 				       parent_real);
 	if (!response) /* whoa, this is bad */ {
 		g_free(parent_real);
@@ -2313,7 +2307,7 @@
 		
 		/* add the dirsep to the end of parent_name */
 		name = g_strdup_printf ("%s%c", parent_real, imap_store->dir_sep);
-		response = camel_imap_command (imap_store, NULL, ex, "CREATE %S",
+		response = camel_imap_command (imap_store, NULL, ex, "CREATE %G",
 					       name);
 		g_free (name);
 		
@@ -2330,7 +2324,7 @@
 	real_name = camel_imap_store_summary_path_to_full(imap_store->summary, folder_name, imap_store->dir_sep);
 	full_name = imap_concat (imap_store, parent_real, real_name);
 	g_free(real_name);
-	response = camel_imap_command (imap_store, NULL, ex, "CREATE %S", full_name);
+	response = camel_imap_command (imap_store, NULL, ex, "CREATE %G", full_name);
 	
 	if (response) {
 		CamelImapStoreInfo *si;
@@ -2493,7 +2487,7 @@
 	present = g_hash_table_new(folder_hash, folder_eq);
 	for (j=0;j<2;j++) {
 		response = camel_imap_command (imap_store, NULL, ex,
-					       "%s \"\" %S", j==1 ? "LSUB" : "LIST",
+					       "%s \"\" %G", j==1 ? "LSUB" : "LIST",
 					       pattern);
 		if (!response)
 			goto fail;
@@ -2954,7 +2948,7 @@
 camel_imap_store_readline (CamelImapStore *store, char **dest, CamelException *ex)
 {
 	CamelStreamBuffer *stream;
-	char linebuf[1024];
+	char linebuf[1024] = {0};
 	GByteArray *ba;
 	ssize_t nread;
 	
