diff -ru evolution-data-server-1.6.1.orig/camel/providers/imap/camel-imap-store.c evolution-data-server-1.6.1/camel/providers/imap/camel-imap-store.c
--- evolution-data-server-1.6.1.orig/camel/providers/imap/camel-imap-store.c	2006-01-16 21:25:58.000000000 +0100
+++ evolution-data-server-1.6.1/camel/providers/imap/camel-imap-store.c	2006-04-28 15:51:07.000000000 +0200
@@ -978,16 +978,11 @@
 		camel_exception_clear (ex);
 		ai = camel_getaddrinfo(service->url->host, port, &hints, ex);
 	}
+	
 	if (ai == NULL)
 		return FALSE;
 	
-	if (!(ret = connect_to_server (service, ai, mode, ex)) && mode == MODE_SSL) {
-		camel_exception_clear (ex);
-		ret = connect_to_server (service, ai, MODE_TLS, ex);
-	} else if (!ret && mode == MODE_TLS) {
-		camel_exception_clear (ex);
-		ret = connect_to_server (service, ai, MODE_CLEAR, ex);
-	}
+	ret = connect_to_server (service, ai, mode, ex);
 	
 	camel_freeaddrinfo (ai);
 	
diff -ru evolution-data-server-1.6.1.orig/camel/providers/imap4/camel-imap4-store.c evolution-data-server-1.6.1/camel/providers/imap4/camel-imap4-store.c
--- evolution-data-server-1.6.1.orig/camel/providers/imap4/camel-imap4-store.c	2006-01-05 19:29:01.000000000 +0100
+++ evolution-data-server-1.6.1/camel/providers/imap4/camel-imap4-store.c	2006-04-28 15:51:07.000000000 +0200
@@ -458,16 +458,11 @@
 		camel_exception_clear (ex);
 		ai = camel_getaddrinfo (service->url->host, port, &hints, ex);
 	}
+	
 	if (ai == NULL)
 		return FALSE;
 	
-	if (!(ret = connect_to_server (engine, ai, mode, ex)) && mode == MODE_SSL) {
-		camel_exception_clear (ex);
-		ret = connect_to_server (engine, ai, MODE_TLS, ex);
-	} else if (!ret && mode == MODE_TLS) {
-		camel_exception_clear (ex);
-		ret = connect_to_server (engine, ai, MODE_CLEAR, ex);
-	}
+	ret = connect_to_server (engine, ai, mode, ex);
 	
 	camel_freeaddrinfo (ai);
 	
diff -ru evolution-data-server-1.6.1.orig/camel/providers/pop3/camel-pop3-store.c evolution-data-server-1.6.1/camel/providers/pop3/camel-pop3-store.c
--- evolution-data-server-1.6.1.orig/camel/providers/pop3/camel-pop3-store.c	2005-08-31 06:26:04.000000000 +0200
+++ evolution-data-server-1.6.1/camel/providers/pop3/camel-pop3-store.c	2006-04-28 15:51:07.000000000 +0200
@@ -324,16 +324,11 @@
 		camel_exception_clear (ex);
 		ai = camel_getaddrinfo(service->url->host, port, &hints, ex);
 	}
+	
 	if (ai == NULL)
 		return FALSE;
 	
-	if (!(ret = connect_to_server (service, ai, mode, ex)) && mode == MODE_SSL) {
-		camel_exception_clear (ex);
-		ret = connect_to_server (service, ai, MODE_TLS, ex);
-	} else if (!ret && mode == MODE_TLS) {
-		camel_exception_clear (ex);
-		ret = connect_to_server (service, ai, MODE_CLEAR, ex);
-	}
+	ret = connect_to_server (service, ai, mode, ex);
 	
 	camel_freeaddrinfo (ai);
 	
diff -ru evolution-data-server-1.6.1.orig/camel/providers/smtp/camel-smtp-transport.c evolution-data-server-1.6.1/camel/providers/smtp/camel-smtp-transport.c
--- evolution-data-server-1.6.1.orig/camel/providers/smtp/camel-smtp-transport.c	2005-08-31 06:26:04.000000000 +0200
+++ evolution-data-server-1.6.1/camel/providers/smtp/camel-smtp-transport.c	2006-04-28 15:51:07.000000000 +0200
@@ -431,16 +431,11 @@
 		camel_exception_clear (ex);
 		ai = camel_getaddrinfo(service->url->host, port, &hints, ex);
 	}
+	
 	if (ai == NULL)
 		return FALSE;
 	
-	if (!(ret = connect_to_server (service, ai, mode, ex)) && mode == MODE_SSL) {
-		camel_exception_clear (ex);
-		ret = connect_to_server (service, ai, MODE_TLS, ex);
-	} else if (!ret && mode == MODE_TLS) {
-		camel_exception_clear (ex);
-		ret = connect_to_server (service, ai, MODE_CLEAR, ex);
-	}
+	ret = connect_to_server (service, ai, mode, ex);
 	
 	camel_freeaddrinfo (ai);
 	
