diff -ru evolution-data-server-1.4.2.1.orig/libedataserverui/e-name-selector-dialog.c evolution-data-server-1.4.2.1/libedataserverui/e-name-selector-dialog.c
--- evolution-data-server-1.4.2.1.orig/libedataserverui/e-name-selector-dialog.c	2006-03-27 11:26:59.000000000 +0200
+++ evolution-data-server-1.4.2.1/libedataserverui/e-name-selector-dialog.c	2006-03-27 11:34:37.000000000 +0200
@@ -703,10 +703,9 @@
 	EContactStore     *contact_store;
 	EDestinationStore *destination_store;
 	EContact          *contact;
+	GtkTreeIter       iter;
 	Section           *section;
 	gint               email_n;
-	GList		  *rows, *l;
-	GtkTreeSelection  *contact_selection;
 
 	/* When a contact is activated, we transfer it to the first destination on our list */
 
@@ -716,38 +715,28 @@
 	if (name_selector_dialog->sections->len == 0) 
 		return;
 
+	/* Get the contact to be transferred */
+
+	if (!gtk_tree_model_get_iter (GTK_TREE_MODEL (name_selector_dialog->contact_sort),
+				      &iter, path))
+		g_assert_not_reached ();
+
+	sort_iter_to_contact_store_iter (name_selector_dialog, &iter, &email_n);
+
+	contact = e_contact_store_get_contact (contact_store, &iter);
+	if (!contact) {
+		g_warning ("ENameSelectorDialog could not get selected contact!");
+		return;
+	}
+	
 	section = &g_array_index (name_selector_dialog->sections, Section, 0);
 	if (!e_name_selector_model_peek_section (name_selector_dialog->name_selector_model,
 						 section->name, NULL, &destination_store)) {
 		g_warning ("ENameSelectorDialog has a section unknown to the model!");
 		return;
 	}
-
-	contact_selection = gtk_tree_view_get_selection (name_selector_dialog->contact_view);
-	rows = gtk_tree_selection_get_selected_rows (contact_selection, NULL);
-	rows = g_list_reverse (rows);
-
-	/* Get the contacts to be transferred */
-
-	for (l=rows; l; l=g_list_next(l)) {
-		GtkTreeIter iter;
-
-		if (!gtk_tree_model_get_iter (GTK_TREE_MODEL (name_selector_dialog->contact_sort),
-				      &iter, path))
-			g_assert_not_reached ();
-
-		gtk_tree_path_free (path);
-		sort_iter_to_contact_store_iter (name_selector_dialog, &iter, &email_n);
-		
-		contact = e_contact_store_get_contact (contact_store, &iter);
-		if (!contact) {
-			g_warning ("ENameSelectorDialog could not get selected contact!");
-			g_list_free (rows);
-			return;
-		}
-		add_destination (destination_store, contact, email_n);
-	}
-	g_list_free (rows);
+	
+	add_destination (destination_store, contact, email_n);
 }
 
 static void
@@ -758,8 +747,7 @@
 	EDestinationStore *destination_store;
 	EDestination      *destination;
 	Section           *section;
-	GList		  *rows, *l;
-	GtkTreeSelection  *selection;
+	GtkTreeIter       iter;
 
 	/* When a destination is activated, we remove it from the section */
 
@@ -776,25 +764,13 @@
 		return;
 	}
 
-	rows = gtk_tree_selection_get_selected_rows (selection, NULL);
-	rows = g_list_reverse (rows);
-
-	for (l = rows; l; l = g_list_next(l)) {
-		GtkTreeIter iter;
-		GtkTreePath *path = l->data;
-
-		if (!gtk_tree_model_get_iter (GTK_TREE_MODEL (destination_store),
-				      	      &iter, path))
-			g_assert_not_reached ();
-
-		gtk_tree_path_free (path);
-
-		destination = e_destination_store_get_destination (destination_store, &iter);
-		g_assert (destination);
+	if (!gtk_tree_model_get_iter (GTK_TREE_MODEL (destination_store), &iter, path))
+		g_assert_not_reached ();
+	
+	destination = e_destination_store_get_destination (destination_store, &iter);
+	g_assert (destination);
 
-		e_destination_store_remove_destination (destination_store, destination);
-	}
-	g_list_free (rows);
+	e_destination_store_remove_destination (destination_store, destination);
 }
 
 static gboolean 
