commit a2bb87c9e448ebf261513f1dc67d255e4077b062
Author: Yves-Alexis Perez <corsac@debian.org>
Date:   Sun Jan 11 15:57:28 2009 +0100

    don't add libdb/dist and calendar/libical to SUBDIRS

diff --git a/configure.in b/configure.in
index 71b3a72..b1f9c8c 100644
--- a/configure.in
+++ b/configure.in
@@ -1663,12 +1663,6 @@ export privlibdir
 export privincludedir
 export privdatadir
 
-if test $enable_calendar = yes; then
-        AC_CONFIG_SUBDIRS(calendar/libical)
-fi
-if test $dynamic_libdb = no; then
-        AC_CONFIG_SUBDIRS(libdb/dist)
-fi
 AC_OUTPUT([
 Makefile
 evolution-data-server-zip
