--- addressbook/backends/ldap/e-book-backend-ldap.c.orig	2006-05-03 15:59:23.000000000 +0200
+++ addressbook/backends/ldap/e-book-backend-ldap.c	2007-01-11 09:20:40.000000000 +0100
@@ -1020,7 +1020,8 @@
 		ldap_abandon (bl->priv->ldap, op->id);
 	g_static_rec_mutex_unlock (&eds_ldap_handler_lock);
 
-	op->dtor (op);
+	if (op->dtor)
+		op->dtor (op);
 
 	bl->priv->active_ops--;
 
@@ -3851,7 +3852,7 @@
 
 			d(printf ("looked up msgid %d, got op %p\n", msgid, op));
 
-			if (op)
+			if (op && op->handler)
 				op->handler (op, res);
 			else
 				g_warning ("unknown operation, msgid = %d", msgid);
