diff -u -r1.36.2.2 -r1.36.2.3
--- evolution-data-server-1.4.2.1.org/calendar/backends/http/e-cal-backend-http.c	2005/10/27 09:21:52	1.36.2.2
+++ evolution-data-server-1.4.2.1/calendar/backends/http/e-cal-backend-http.c	2006/01/02 12:04:56	1.36.2.3
@@ -127,6 +127,11 @@
 		priv->soup_session = NULL;
 	}
 
+	if (priv->reload_timeout_id) {
+		g_source_remove (priv->reload_timeout_id);
+		priv->reload_timeout_id = 0;
+	}
+
 	g_free (priv);
 	cbhttp->priv = NULL;
 
