diff -ru evolution-data-server-1.4.2.1.orig/libedataserverui/libedataserverui.pc.in evolution-data-server-1.4.2.1/libedataserverui/libedataserverui.pc.in
--- evolution-data-server-1.4.2.1.orig/libedataserverui/libedataserverui.pc.in	2005-11-29 03:03:48.000000000 +0000
+++ evolution-data-server-1.4.2.1/libedataserverui/libedataserverui.pc.in	2006-03-29 12:54:27.000000000 +0000
@@ -11,6 +11,6 @@
 Name: libedataserverui
 Description: GUI utility library for evolution data server
 Version: @VERSION@
-Requires: libbonobo-2.0 >= @LIBBONOBO_REQUIRED@ libgnome-2.0 libxml-2.0 ORBit-2.0 >= @ORBIT_REQUIRED@ libedataserver-1.2
+Requires: gtk+-2.0 libbonobo-2.0 >= @LIBBONOBO_REQUIRED@ libgnome-2.0 libxml-2.0 ORBit-2.0 >= @ORBIT_REQUIRED@ libedataserver-1.2
 Libs: -L${libdir} -ledataserverui-1.2
 Cflags: -I${privincludedir}
