diff -ru evolution-data-server-1.6.2.orig/camel/providers/imap/camel-imap-folder.c evolution-data-server-1.6.2/camel/providers/imap/camel-imap-folder.c
--- evolution-data-server-1.6.2.orig/camel/providers/imap/camel-imap-folder.c	2006-05-22 08:59:10.000000000 +0200
+++ evolution-data-server-1.6.2/camel/providers/imap/camel-imap-folder.c	2006-05-30 20:06:12.000000000 +0200
@@ -2308,7 +2308,7 @@
 }
 
 
-#define CAMEL_MESSAGE_INFO_HEADERS "DATE FROM TO CC SUBJECT REFERENCES IN-REPLY-TO MESSAGE-ID MIME-VERSION CONTENT-TYPE"
+#define CAMEL_MESSAGE_INFO_HEADERS "DATE FROM TO CC SUBJECT REFERENCES IN-REPLY-TO MESSAGE-ID MIME-VERSION CONTENT-TYPE "
 
 /* FIXME: this needs to be kept in sync with camel-mime-utils.c's list
    of mailing-list headers and so might be best if this were
