--- camel-imap-folder.c.old	2006-05-10 11:42:24.000000000 +0200
+++ evolution-data-server/camel/providers/imap/camel-imap-folder.c	2006-05-10 11:42:10.000000000 +0200
@@ -2322,7 +2322,7 @@
 }
 
 
-#define CAMEL_MESSAGE_INFO_HEADERS "DATE FROM TO CC SUBJECT REFERENCES IN-REPLY-TO MESSAGE-ID MIME-VERSION CONTENT-TYPE"
+#define CAMEL_MESSAGE_INFO_HEADERS "DATE FROM TO CC SUBJECT REFERENCES IN-REPLY-TO MESSAGE-ID MIME-VERSION CONTENT-TYPE "
 
 /* FIXME: this needs to be kept in sync with camel-mime-utils.c's list
    of mailing-list headers and so might be best if this were
