---
 CMakeLists.txt |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -127,7 +127,7 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 # Packagers might want to need different settings for the RPATH related things
 # From experience, especially CMAKE_BUILD_WITH_INSTALL_RPATH might need to be
 # switched to ON, if CMake fails to set the right values during make install
-set(CMAKE_SKIP_RPATH OFF)
+set(CMAKE_SKIP_RPATH ON)
 set(CMAKE_SKIP_BUILD_RPATH OFF CACHE BOOL INTERNAL)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH OFF CACHE BOOL INTERNAL)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON CACHE BOOL INTERNAL)
