--- camel/camel-mime-utils.c.orig	2005-04-14 17:25:21.000000000 +0900
+++ camel/camel-mime-utils.c	2005-04-14 18:17:05.000000000 +0900
@@ -1116,8 +1116,29 @@
 		}
 
 		start = inptr;
-		while (inptr < inend && !camel_mime_is_type (*inptr, mask))
+
+		/* modified to decode not space-separated mime header */
+		while (inptr < inend && !camel_mime_is_type (*inptr, mask)) {
+			if (*start == '=' && *(start + 1) == '?' &&
+			    *inptr == '?' && *(inptr + 1) == '=') {
+				/* =?UTF-8?Q?=... should be continue
+				 * or not should be break;
+				 */
+				if (*(inptr - 2) != '?') {
+					/* end of MIME */
+					inptr++;
+					inptr++;
+					break;
+				}
+			} else if ((*start != '=' || *(start + 1) != '?') &&
+				   *(inptr + 1) == '=' && *(inptr + 2) == '?') {
+				/* 'P=?...' */
+				inptr++;
+				/* will be start of MIME, maybe */
+				break;
+			}
 			inptr++;
+		}
 
 		dword = rfc2047_decode_word(start, inptr-start);
 		if (dword) {
