--- evolution/camel/providers/pop3/camel-pop3-store.c.orig	2003-06-17 16:22:48.000000000 +0900
+++ evolution/camel/providers/pop3/camel-pop3-store.c	2003-06-17 16:25:38.000000000 +0900
@@ -589,7 +589,7 @@
 	
 	/* Now that we are in the TRANSACTION state, try regetting the capabilities */
 	store->engine->state = CAMEL_POP3_ENGINE_TRANSACTION;
-	camel_pop3_engine_reget_capabilities (store->engine);
+	/* camel_pop3_engine_reget_capabilities (store->engine); */
 	
 	return TRUE;
 }
